### DISCUSSION TITLE
Explanation of checked vs unchecked inputs?

### DISCUSSION THREAD

**Discussion by dannypaz** [prisma/prisma#14313] [Q&A]
Im looking for more documentation on typescript types w/ checked vs unchecked inputs and what the differences are?

Im noticing in my current code that using `upsert` with a `where` clause causes typechecking to fail unless I typecase my values with UncheckedInput (assuming this is cause the id of the record is inferred). I'd love to have more information on this

**Comment by TasinIshmam** [comment]
Hi @dannypaz  

[This answer](https://github.com/prisma/prisma/discussions/10121#discussioncomment-1621254) explains the basic difference between checked and unchecked inputs. 


>  Im noticing in my current code that using upsert with a where clause causes typechecking to fail unless I typecase my values with UncheckedInput (assuming this is cause the id of the record is inferred). I'd love to have more information on this

If you provide your code/schema or a simple reproduction, I'd be happy to look into this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
